subgraphs/uniswap-forks/protocols/camelot-v2/src/price.ts

```diff
@@ -0,0 +1,364 @@
+          event,
+          pool.inputTokens[whitelistTokenIndex],
+          false
+        );
+        const whitelistTokenLocked = poolAmounts.inputTokenBalances[
+          whitelistTokenIndex
+        ].times(whitelistToken.lastPriceUSD!);
+        if (
+          whitelistTokenLocked.gt(largestLiquidityWhitelistTokens) &&
+          whitelistTokenLocked.gt(
+            NetworkConfigs.getMinimumLiquidityThresholdTrackPrice()
+          )
+        ) {
+          largestLiquidityWhitelistTokens = whitelistTokenLocked;
+
+          // Check whether current pool is a stable pair using a different formula
+          const pair = Pair.bind(Address.fromString(pool.id));
+          const stable = pair.stableSwap();
+
+          let newPrice = BIGDECIMAL_ZERO;
+          if (stable) {
+            newPrice = calculateRateForStablePair(
+              pool,
+              pair,
+              token,
+              whitelistToken
+            ).times(whitelistToken.lastPriceUSD!);
+          } else {
+            newPrice = safeDiv(
+              poolAmounts.inputTokenBalances[whitelistTokenIndex],
+              poolAmounts.inputTokenBalances[priceTokenIndex]
+            ).times(whitelistToken.lastPriceUSD! as BigDecimal);
+          }
+
+          if (isValidTVL(pool, token, newPrice)) {
+            priceSoFar = newPrice;
+          }
+        }
+      }
+    }
+  }
+
+  if (priceSoFar.equals(BIGDECIMAL_ZERO)) {
+    return priceSoFar;
+  }
+
+  // Buffer token pricings that would cause large spikes on the protocol level
+  const protocol = getOrCreateProtocol();
+  const tokenTVLDelta = absBigDecimal(
+    priceSoFar
+      .times(convertTokenToDecimal(token._totalSupply, token.decimals))
+      .minus(token._totalValueLockedUSD)
+  );
+  const protocolTVLPercentageDelta = absBigDecimal(
+    safeDiv(tokenTVLDelta, protocol.totalValueLockedUSD)
+  );
+  if (protocolTVLPercentageDelta.gt(BIGDECIMAL_FIVE_PERCENT)) {
+    log.warning("Price too high for token: {} from pool: {}", [
+      token.id,
+      priceSoFar.toString(),
+    ]);
+    if (token._largeTVLImpactBuffer < PRICE_CHANGE_BUFFER_LIMIT) {
+      token._largeTVLImpactBuffer += 1;
+      token.save();
+      return token.lastPriceUSD!;
+    }
+  }
+
+  if (!token.lastPriceUSD || token.lastPriceUSD!.equals(BIGDECIMAL_ZERO)) {
+    token.save();
+    return priceSoFar;
+  }
+
+  // If priceSoFar 10x greater or less than token.lastPriceUSD, use token.lastPriceUSD
+  // Increment buffer so that it allows large price jumps if seen repeatedly
+  if (
+    priceSoFar.gt(token.lastPriceUSD!.times(BIGDECIMAL_TWO)) ||
+    priceSoFar.lt(token.lastPriceUSD!.div(BIGDECIMAL_TWO))
+  ) {
+    if (token._largePriceChangeBuffer < PRICE_CHANGE_BUFFER_LIMIT) {
+      token._largePriceChangeBuffer += 1;
+      token.save();
+      return token.lastPriceUSD!;
+    }
+  }
+
+  token._largePriceChangeBuffer = 0;
+  token._largeTVLImpactBuffer = 0;
+
+  token.save();
+  return priceSoFar;
```

Comment:
should save `token.lastPriceUSD = priceSoFar` before returning

Reply:
Added, but this is copied from the common src.

This also works as is, since findUSDPricePerToken only has one usage and that saves token.lastPriceUSD.